tests/nn_test.py

```diff
@@ -720,6 +720,28 @@ def f(hx, _):
     with jax.checking_leaks():
       fwd()  # doesn't crash
 
+  def testLog1mExp(self):
+    x, expected = jnp.array([
+        [0.1, jnp.log(1 - jnp.exp(-0.1))],
+        [1.1, jnp.log(1 - jnp.exp(-1.1))],
+        [0, -jnp.inf],
+        [1, -0.45867515],
+        [1e2, 0.0],
+        [1e-5, jnp.log(1e-5)],
+        [-1, jnp.nan],
+        [-1e-2, jnp.nan],
+        [-1e2, jnp.nan],
+        [jnp.inf, 0.0],
+    ]).T
+    got = nn.log1mexp(x)
+    self.assertAllClose(got, expected)
+
+  def testLog1mExpGrad(self):
```

Comment:
Now that we're using `custom_jvp`, this test is not as useful (it's basically just an assertion of an identity)

For functions that use `custom_jvp`, we should probably test them with `jtu.check_grads` to check the logic of the custom autodiff rule.